src/window.rs

```diff
@@ -42,29 +44,36 @@ fn handle_new_grid_size(new_size: LogicalSize, renderer: &Renderer) {
     }
 }
 
-struct WindowWrapper {
-    context: Sdl,
-    window: Window,
-    skulpin_renderer: SkulpinRenderer,
-    renderer: Renderer,
-    mouse_down: bool,
-    mouse_position: LogicalSize,
-    title: String,
-    previous_size: LogicalSize,
-    previous_dpis: (f32, f32),
-    transparency: f32,
-    fullscreen: bool,
+pub struct WindowWrapper {
+    pub context: Sdl,
+    pub window: Option<Window>,
```

Comment:
The introduction of the option here is rough because it forces any usage of the window to be wrapped in an if statement. Can you help me understand what the state with no available window looks like? Pretty sure neovide wouldn't work very well without a rendering surface or input?

Reply:
I wanted to keep neovide open so that I could create new windows without having to run the other initialization steps again, so I wanted to keep neovide alive and close the window. The cleanest way to do that was to allow rust to just drop the window. I could have also invoked SDL through sys to drop the window.

Reply:
This is related to my comment below. I wonder if a better way of achieving this would be to implement your own window, and send UICommands to the bridge, and do equivalent draw logic in your app. Then whenever the editor component resizes, you get imput events etc you filter and pass the relevant ones to neovide's systems. This way neovide doesn't have to have more complexity, and you get full control over the renderer etc.

Reply:
Totally down for that. Though would this entail a full reengineering? Are you thinking of using sender and receiver logic to decouple the event handling from the renderer? I also had some ideas about using a texture instead of a full window, but the windowing is nice to have.

Reply:
Today I believe that the editor, bridge, and renderer components don't depend on the window. They are consumed by the window system. You'd have to pull in skribo (or produce your own skia-safe canvas) so that you can pass  a canvas to the renderer draw function. Similarly youd have to push ui events to the bridge and pull draw commands from the editor, but neither of those require anything specific to the window

Reply:
One bit thats somewhat weird is that the bridge and editor components are hard coded singletons. So maybe an action item on my side that I wouldn't mind doing would be to remove the singleton relationship. That way you could initialize a new neovide process and spin it down multiple times over the course of your app's lifetime

Reply:
That sounds great. I'm stepping back and looking at the WindowWrapper, and I think it can be converted into a trait and impl.

Reply:
Feel free to ping me with questions or w/e. I can respond here or on the gitter chat if you prefer.

Reply:
This also eliminates the need for making public struct members.

Reply:
Totally! I think its a more principled change overall.

Reply:
Note: I'm not proposing that we change the windowmanager into a trait because it isn't used by anything. In neovide today, the windowmanager uses and manages everything else. So your app would implement something similar, but doesn't need to use a unified trait because it wouldn't be consumed by anything on the neovide side

Reply:
The only interface is the skia-safe renderer which uses the sdl2 window. Other than that, everything goes from the window manager out to the other components. Not the other way around.

Reply:
That sounds like a good entrypoint for anyone who wants to sort of plug into the other components. Not sure if we're in agreement on whether to make a WindowWrapper component or not. As a person who is more of a "customer", I think I would find a window trait to be useful in helping cover the full functionality for text editing. If the trait is too restrictive, then I could just use the individual components.

Reply:
Its not really about being restrictive so much as that trait wouldn't be used by anything. Its very possible I'm misunderstanding your idea though so don't let that stop you :)